ref: Update `dvc run` references in `dvc commit` (#3035)
iterative/dvc.org: content/docs/command-reference/commit.md

- Store the file contents in the cache.

> Skipping these steps is typically done to avoid caching unfinished data, for

> > example when [experimenting](/doc/command-reference/exp).

Reviewer: ```suggestion
> example when [experimenting](/doc/user-guide/experiment-management).
```

Reviewer: But I'm not sure we want to plug experiments here. I don't remember `commit` used much if at all in any exp ref or guide. Probably roll back

Author: They'll need when  using `--no-commit` flag to `dvc exp run`. For the singular use, `exp run` is no different from `dvc repro`, it leaves artifacts in the workspace and those may be committed or not.

Reviewer: It's different. It creates an experiment (Git custom ref. etc.) If this is not about experimentation let's use non-exp concepts.

Reviewer 2: Agreed, it's also about `dvc add` which might be used outside experiments.

Author: I replace `repro` with `exp` here, because the sentence mentions "unfinished data." Usually we have "unfinished data" while experimenting. I can add "or [running pipelines] (repro)" here, or remove this sentence completely. 

`dvc add` is mentioned in the paragraph one above.

Reviewer: > Usually we have "unfinished data" while experimenting.

Sure, "experimenting" in a very general sense, as in "trying things out". That's still not the same as a data science experiment (param tuning, etc.)

> I can add "or [running pipelines] (repro)" here, or remove this sentence completely.I don't think it's a problem to mention "unfinished data" here without associating it with `dvc exp`.

>

Reviewer: OOPS FIXING PREVIOUS COMMENT:

> I can add "or [running pipelines] (repro)" here, or remove this sentence completely.

I don't think it's a problem to mention "unfinished data" here without associating it with `dvc exp`. And not sure I see how running pipelines relates exactly to `--no-commit` -- you can still run it right? But again that's a jump away from the topic I think. We can just roll this link back IMO.

Author: Removed it. PTAL. @jorgeorpinel

Reviewer: Thanks.

> Skipping these steps is typically done to avoid caching unfinished data, for
> example when exploring different data or

> > [stages](/doc/command-reference/stage)

Reviewer: There's also `<abbr>stage</abbr>` - probably better?

Author: Done. Thx.

Sometimes we want to iterate through multiple changes to configuration, code, or
data, trying different ways to improve the output of a stage. To avoid filling

> the <abbr>cache</abbr> with undesired intermediate results, we can run a single

Reviewer: ```suggestion
the <abbr>cache</abbr> with undesired intermediate results, you can
```

> stage, or reproduce an entire pipeline by supplying `--no-commit` option to
> `dvc repro`. This prevents data from being pushed to the cache. When development
> of stage is finished, `dvc commit` can be used to store data files in the cache.

Reviewer: ```suggestion
use the `--no-commit` option of `dvc repro`. Once your progress is good
enough, `dvc commit` can be used to store data files in the cache.
```

Sometimes we want to iterate through multiple changes to configuration, code, or
data, trying different ways to improve the output of a stage. To avoid filling
the <abbr>cache</abbr> with undesired intermediate results, we can run a single

> stage with `dvc exp run --no-commit`, or reproduce an entire pipeline using

Reviewer: Idk. This produces an experiment which also fills the cache. Should be `repro` I think.

Author: Which cache? Run cache or the file cache? `dvc exp run` command works similar to repro, commits artifacts to `.dvc/cache`

Reviewer: Regardless of the cache question, `exp run` produces an experiment. We have not decided to use experiment runs as the default pipeline workflow that I know of so any non-experiment specific docs should not use exp commands. Thus `repro` here.

Reviewer 2: Agreed, this is not a valid change semantically.

Probably `dvc stage add` and then `dvc repro --no-commit <stage>` is the way to replace it.

Or ideally `dvc stage add` should have `--execute`.

Reviewer: > Probably dvc stage add and then dvc repro --no-commit <stage> is the way to replace it.

Agree, emphasizing the `<stage>` given as `target`.

> I removed the "entire pipeline" phrase

Why? It's a valid situation we want to mention. If we're going to change that, rather generalize to `repro --no-commit` only and don't mention either specific-stage(s) nor entire-pipeline cases, as that distinction is besides the point, really.

Author: Restored the phrase now. PTAL

Sometimes we want to iterate through multiple changes to configuration, code, or
data, trying different ways to improve the output of a stage. To avoid filling
the <abbr>cache</abbr> with undesired intermediate results, we can run a single
stage with `dvc repro --no-commit <target>`, or reproduce an entire pipeline

> with `dvc repro --no-commit`. This prevents data from being pushed to the cache.

Reviewer: I'd combine these into a single one (probably no need to differentiate between single or multiple stages) but OK, not the direct purpose of this PR.

Author: I've merged them, could you PTAL?

Reviewer: Thanks, I'll leave a suggestion that further generalizes this. ⌛
